src/utilities.ts

```diff
@@ -54,6 +56,10 @@ export type ResultMatcherNoReturn<TValue, TError> = {
   failure: ActionOfT<TError>;
 };
 
+export type ResultValueOf<T> = T extends Result<infer TResultValue>
```

Comment:
Currently `utilities.ts` is types and pure functions only (no dependencies on the monads). Do you think `ResultValueOf<T>` should go in here even though it is dependent on `Result`?

What about moving it to `result.ts`?

Reply:
By the way, I like this type a lot! I still don't understand how `infer` works even though I've read about it several times 😵.

Reply:
## utilities.ts

This is a good point.
Currently, there is a cyclic file dependency between result.ts and utiltities.ts. I don't like this, either.
I will move `ResultValueOf` back to result.ts/

## infer

Using Generics, the generic parameter is set by the developer: `Result<string>, Result<Customer>`.
The outcome of Result depends on what the developer gives us.

`infer` allows us to look into the Generic and extract the passed parameter.

In our case, it helps us, to look into the `Record<string, Result<???>>` that `.combine` processes.
Each generic parameter gets extracted with `infer` and is remapped to our new Result containing all values.